src/main/java/net/dv8tion/jda/api/entities/Entitlement.java

```diff
@@ -0,0 +1,135 @@
+ */
+public interface Entitlement extends ISnowflake
+{
+    /**
+     * The id of the entitlement
+     *
+     * @return The id of the {@link Entitlement Entitlement}
+     */
+
+    @Override
+    long getIdLong();
+
+    /**
+     * The id of the sku related to this {@link Entitlement Entitlement}
+     *
+     * @return The id of the SKU related to this {@link Entitlement Entitlement}
+     */
+    long getSkuIdLong();
+
+    /**
+     * The id of the sku related to this {@link Entitlement Entitlement}
+     *
+     * @return The id of the SKU related to this {@link Entitlement Entitlement}
+     */
+    @Nonnull
+    String getSkuId();
+
+    /**
+     * The id of the parent application of this {@link Entitlement Entitlement}
+     *
+     * @return The id of the parent application of this {@link Entitlement Entitlement}
+     */
+    long getApplicationIdLong();
+
+    /**
+     * The id of the parent application of this {@link Entitlement Entitlement}
+     *
+     * @return The id of the parent application of this {@link Entitlement Entitlement}
+     */
+    @Nonnull
+    String getApplicationId();
+
+    /**
+     * The id of the user that purchased the {@link Entitlement Entitlement}
+     *
+     * @return The id of the user that purchased the {@link Entitlement Entitlement}
+     */
+    @Nullable
+    Long getUserIdLong();
+
+    /**
+     * The id of the user that purchased the {@link Entitlement Entitlement}
+     *
+     * @return The id of the user that purchased the {@link Entitlement Entitlement}
+     */
+    @Nullable
+    String getUserId();
+
+    /**
+     * The guild id that is granted access to the {@link Entitlement Entitlement}'s sku
+     *
+     * @return The id of the guild that is granted access to the {@link Entitlement Entitlement}'s sku,
+     * or Null if this entitlement is related to a subscription of type "User Subscription"
+     */
+    @Nullable
+    Long getGuildIdLong();
+
+    /**
+     * The guild id that is granted access to the {@link Entitlement Entitlement}'s sku
+     *
+     * @return The id of the guild that is granted access to the {@link Entitlement Entitlement}'s sku,
+     * or Null if this entitlement is related to a subscription of type "User Subscription"
+     */
+    @Nullable
+    String getGuildId();
+
+    /**
+     * The type of the Entitlement
+     * <br>The only possible value of this property is 8 at the moment and indicates the "APPLICATION_SUBSCRIPTION" type
+     * <br>Discord doesn't currently support other types for entitlements.
+     *
+     * @return the {@link Entitlement Entitlement} type, 8 is the only possible value
+     */
+    int getType();
+
+    /**
+     * Whether the {@link Entitlement Entitlement} has been deleted or not.
+     *
+     * @return True if the {@link Entitlement Entitlement} was deleted, False otherwise
+     *
+     * @see net.dv8tion.jda.api.events.entitlement.EntitlementDeleteEvent
```

Comment:
```suggestion
     * @return True if the {@link Entitlement Entitlement} was deleted, False otherwise
     *
     * @see     net.dv8tion.jda.api.events.entitlement.EntitlementDeleteEvent
```